paper.md

```diff
@@ -318,19 +326,42 @@ The training conditions were exactly similar to the previous two case studies.
 
-# Results {#sec:result}
+### Setting up CI for our project
+
+We pushed our MNIST project to Github and activated the repository in Travis to set up continuous integration on our project.
+For long-running workflows like those consisting of ML/AI or BigData workloads, it is recommended to scale down various parameters like dataset size, epochs, etc. with the help of environment variables to reduce the CI running time and iterate quickly. 
+We declared environment variables like `NUM_EPOCHS`, `DATASET_REDUCTION`, and `BATCH_SIZE` to control the number of epochs, size of training data, and batch size respectively in our workflow.
+Using the above variables we used only 10% of the dataset and configured the training for a single epoch, thus effectively reducing our CI running time by approx. 75%.
+The `.travis.yml` file used by our case study is shown below. 
+It can be generated by running `popper ci travis` from the command line.
 
-A summary of the training duration and accuracy obtained by running the workflow on three different computing environment is shown below.
+```yaml
+---
+dist: xenial
+language: python
+python: 3.7
+services: docker
+install:
+- git clone https://github.com/systemslab/popper /tmp/popper
+- export PYTHONUNBUFFERED=1
+- pip install /tmp/popper/src
+script: popper run -f artifacts/local/.popper.yml -w artifacts/local
+```
 
-| Compute Environment    | Avg. Duration | Avg. Accuracy    |
-|------------------------|---------------|------------------|
-| Local Machine          |     4034.2s   |      99.16%      |
-| Kubernetes Cluster     |     1001.6s   |      99.236%     | 
-| Slurm Cluster          |     234.8s    |      98.69%      |
+By setting up CI, users can continuously validate changes made to their workflows and also protect their workflows from getting outdated due to reasons such as outdated dependencies, outdated container images, broken links, etc.
+
+# Results {#sec:result}
 
 ![Comparison of training durations in 3 different computing environments with Popper](./figures/plot.png){#fig:casestudies}
 
-From Table I & Figure III, it can be seen how the portability of Popper workflows drastically reduces software development and debugging time by enabling developers and researchers to quickly iterate and test in different computing environments.
+A summary of the training duration and accuracy obtained by running the workflow in three different computing environment is shown in Figure @Fig:casestudies.
+It can be seen from the graph how the portability of Popper workflows drastically reduces software development and debugging time by enabling developers and researchers to quickly iterate and test in different computing environments.
+
+![Popper configuration files for different computing environments](./figures/config.jpg){#fig:conffiles}
```

Comment:
nice! Would it look better if we add those verbatim? Also, I think we can skip the one for the local machine, as `docker` is the default.

Reply:
OK